tests/demo_native.rs

```diff
@@ -0,0 +1,250 @@
+        })
+    }
+
+    /// Number of blocks to wait before deeming the test successful
+    fn expected_block_height(&self) -> u64 {
+        self.expected_block_height
+    }
+
+    async fn latest_light_client_update(&self) -> u64 {
+        let filter = Filter::new()
+            .from_block(BlockNumber::Earliest)
+            .address(self.light_client_address);
+        // block number for latest light client update
+        let latest_light_client_update = self
+            .provider
+            .get_logs(&filter)
+            .await
+            .unwrap()
+            .last()
+            .unwrap()
+            .block_number;
+
+        latest_light_client_update.unwrap().as_u64()
+    }
+    async fn test_state(&self) -> TestState {
+        let block_height = self.espresso.get_height().await.ok();
+        let txn_count = self.espresso.get_transaction_count().await.unwrap();
+
+        let builder_balance = self
+            .espresso
+            .get_espresso_balance(self.builder_address, block_height)
+            .await
+            .unwrap();
+        let recipient_balance = self
+            .espresso
+            .get_espresso_balance(self.recipient_address, block_height)
+            .await
+            .unwrap();
+
+        let light_client_update = self.latest_light_client_update().await;
+
+        TestState {
+            block_height,
+            txn_count,
+            builder_balance,
+            recipient_balance,
+            light_client_update,
+        }
+    }
+    async fn readiness(&self) -> anyhow::Result<Vec<String>> {
+        join_all(vec![
+            wait_for_service(&self.load_generator_url, 1000, 600),
+            wait_for_service(&self.builder_url, 1000, 60),
+            wait_for_service(&self.prover_url, 1000, 300),
+        ])
+        .await
+        .into_iter()
+        .collect::<anyhow::Result<Vec<String>>>()
+    }
+}
+
+#[async_std::test]
+async fn test_smoke() -> anyhow::Result<()> {
+    let start = Instant::now();
+    dotenvy::dotenv()?;
+
+    let testing = TestConfig::new().await.unwrap();
+    let _ = testing.readiness().await?;
+
+    let initial = testing.test_state().await;
+    println!("Initial State:{}", initial);
+
+    loop {
+        sleep(Duration::from_secs(1)).await;
+
+        let new = testing.test_state().await;
+
+        println!("New State:{}", new);
+        if new <= initial {
+            panic!("Chain state not incrementing");
+        }
+
+        if initial.builder_balance + initial.recipient_balance
+            != new.builder_balance + new.recipient_balance
+        {
+            panic!("Balance not conserved");
+        }
+
+        if start.elapsed().as_secs() > SEQUENCER_BLOCKS_TIMEOUT {
+            panic!("Timeout waiting for block height, transaction count, and light client updates to increase.");
+        }
+
+        // test that we progress EXPECTED_BLOCK_HEIGHT blocks from where we started
+        if new.block_height.unwrap()
```

Comment:
I think we should check multiple times that block height is increasing like maybe for 1-2minutes.